src/idrs.jl

```diff
@@ -80,16 +81,23 @@ idrs(A, b; s = 8, tol=sqrt(eps(typeof(real(b[1])))), maxiter = length(b)^2) =
 idrs!(x, A, b; s = 8, tol=sqrt(eps(typeof(real(b[1])))), maxiter=length(x)^2) =
     idrs_core!(x, linsys_op, (A,), b, s, tol, maxiter)
 
-function idrs_core!{T}(X, op, args, C::T, s, tol, maxiter)
+function idrs_core!{T}(X::T, op, args, C::T, s::Int64, tol::Float64, maxiter::Int64; smoothing::Bool=false)

```

Comment:
``X::T`` is overspecialization. ``C`` is of type ``op(X, args...)``, so X and C not necessarily of the same type.